pkg/version/BUILD.bazel

```diff
@@ -0,0 +1,12 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
```

Comment:
Could this all be just vendored from k/k and re-used? By this I mean everything in this directory.

Reply:
Potentially, but I don't think it would be easy. There are some modifications that I've made to the files to better support the cluster registry:

- changing the package in `def.bzl`
- simplifying the default variable definitions in `base.go`
- not using the `gitMajor` and `gitMinor` flags, which are said to be deprecated

I also don't particularly want to tie the cluster registry to the way that Kubernetes does its versioning, and would rather have room to diverge as needed. Also, vendoring in k/k will potentially make cloning/updating deps/etc more cumbersome, and I'd rather avoid having the cluster registry depend on k/k directly.

If you think there's value in a consistent versioning model across the k8s projects, I can file (or find and +1) an issue that tracks extracting this version functionality into a shared location. I only need the bazel-based process, though, and I expect there would be a lot of k/k-specific stuff that would be hard to usefully extract.

Reply:
What you are saying makes sense. I don't think we should make versioning consistent across k8s projects, but just share the infrastructure if they are exactly identical. If you are planning to diverge, then keep them separate makes sense.